modules/test/playwright/tests/commerce/commerceSpecificationFacet.spec.ts

```diff
@@ -0,0 +1,226 @@
+) => {
+	const postProduct = {
+		active: true,
+		catalogId,
+		name: {
+			en_US: productName,
+		},
+		productSpecifications: [
+			{
+				specificationKey,
+
+				value: {
+					en_US: productName,
+				},
+			},
+		],
+		productStatus: 0,
+		productType: 'simple',
+	};
+
+	return postProduct;
+};
+
+const getSpecification = (
+	optionCategoryId: number,
+	optionCategoryName: number,
+	priority: number
+) => {
+	const postSpecification = {
+		facetable: true,
+		key: optionCategoryName,
+		optionCategory: {
+			id: optionCategoryId,
+			key: optionCategoryName,
+			priority,
+			title: {
+				en_US: optionCategoryName,
+			},
+		},
+		title: {
+			en_US: optionCategoryName,
+		},
+	};
+
+	return postSpecification;
+};
+
+test('can sort specifications by specification group priority', async ({
+	_apiHelpers,
+	_layoutsPage,
+	_specificationFacetsPage,
+}) => {
+	const pageLabel = 'Specification Facet Page';
+
+	await _layoutsPage.createWidgetPage(pageLabel);
+
+	await expect(
+		_layoutsPage.page.getByRole('heading', {
+			name: pageLabel,
+		})
+	).toBeVisible({timeout: 60 * 1000});
+
+	await _specificationFacetsPage.addSearchOptionsWidget();
+	await _specificationFacetsPage.addSpecificationFacetWidget();
+
+	const site = await _apiHelpers.headlessAdminUser.getSiteByFriendlyUrlPath(
+		'guest'
+	);
+
+	const channel = await _apiHelpers.headlessCommerceAdminChannel.postChannel(
+		getChannel('Specification Facet Channel', site.id)
+	);
+
+	const optionCategory1 = await _apiHelpers.headlessCommerceAdminCatalog.postOptionCategory(
+		getOptionCategory('Warranty', 0)
+	);
+	const optionCategory2 = await _apiHelpers.headlessCommerceAdminCatalog.postOptionCategory(
+		getOptionCategory('Material', 1)
+	);
+
+	const specification1 = await _apiHelpers.headlessCommerceAdminCatalog.postSpecification(
+		getSpecification(
+			optionCategory1.id,
+			optionCategory1.key,
+			optionCategory1.priority
+		)
+	);
+	const specification2 = await _apiHelpers.headlessCommerceAdminCatalog.postSpecification(
+		getSpecification(
+			optionCategory2.id,
+			optionCategory2.key,
+			optionCategory2.priority
+		)
+	);
+
+	const catalog = await _apiHelpers.headlessCommerceAdminCatalog.postCatalog(
+		getCatalog('Specification Facet Catalog')
+	);
+
+	const product1 = await _apiHelpers.headlessCommerceAdminCatalog.postProduct(
+		getProduct(catalog.id, 'Product1', specification1.key)
+	);
+	const product2 = await _apiHelpers.headlessCommerceAdminCatalog.postProduct(
+		getProduct(catalog.id, 'Product2', specification2.key)
+	);
+
+	await _specificationFacetsPage.configureSearchOptions();
+
+	await _specificationFacetsPage.goToPage();
+
+	await expect(
+		_layoutsPage.page.getByRole('heading', {
+			name: pageLabel,
+		})
+	).toBeVisible();
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory1.key + '1').isVisible()
+	).toBeTruthy();
+
+	await expect(
+		_layoutsPage.page.getByTestId(optionCategory2.key + '2').isVisible()
+	).toBeTruthy();
+
```

Comment:
I was expecting to see some assert on ordering, this looks like it's only checking the presence of these two option categories

Reply:
@andrea-ale-sbarra , it is my fault for causing the confusion here. I accidentally did not commit the `data-qa-id` [implementation](https://github.com/liferay-commerce/liferay-portal/blob/6f69cbf5031f09758b7e09fcd9a7e027397804fb/modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/specification_option_facets/view.jsp#L89) on the specification facet that would make this work as intended.


I also updated the `expect` checks to be stricter.

```
	await expect(
		_layoutsPage.page.getByTestId(optionCategory1.key + '1')
	).toBeVisible({timeout: 60 * 1000})

	await expect(
		_layoutsPage.page.getByTestId(optionCategory2.key + '2')
	).toBeVisible({timeout: 60 * 1000});
```